test/functional/rpc_psbt.py

```diff
@@ -189,10 +189,46 @@ def run_test(self):
         res = self.nodes[1].walletcreatefundedpsbt(inputs, addr, 0, {"feeRate": 0.1, "add_inputs": True})
         assert_approx(res["fee"], 0.055, 0.005)
 
+        self.log.info("Test passing walletcreatefundedpsbt explicit feerate with conf_target and estimate_mode")
+        for unit, feerate in {"btc/kb": 0.1, "sat/b": 10000}.items():
+            fee = self.nodes[1].walletcreatefundedpsbt(inputs, addr, 0, {"conf_target": feerate, "estimate_mode": unit, "add_inputs": True})["fee"]
+            self.log.info("- conf_target {}, estimate_mode {} produces fee {} at or slightly below -maxtxfee (~0.05290000)".format(feerate, unit, fee))
+            assert_approx(fee, vexp=0.055, vspan=0.005)
+
+        for field, feerate in {"conf_target": 0.1, "estimate_mode": "sat/b"}.items():
+            self.log.info("Test walletcreatefundedpsbt raises RPC error if both feeRate and {} are passed".format(field))
+            assert_raises_rpc_error(
+                -8, "Cannot specify both {} and feeRate".format(field),
+                lambda: self.nodes[1].walletcreatefundedpsbt(inputs, addr, 0, {"feeRate": 0.1, field: feerate, "add_inputs": True}))
+
+        self.log.info("Test walletcreatefundedpsbt with invalid estimate_mode settings")
+        for k, v in {"number": 42, "object": {"foo": "bar"}}.items():
+            assert_raises_rpc_error(-3, "Expected type string for estimate_mode, got {}".format(k),
+                lambda: self.nodes[1].walletcreatefundedpsbt(inputs, addr, 0, {"estimate_mode": v, "conf_target": 0.1, "add_inputs": True}))
+        for mode in ["foo", Decimal("3.141592")]:
+            assert_raises_rpc_error(-8, "Invalid estimate_mode parameter",
+                lambda: self.nodes[1].walletcreatefundedpsbt(inputs, addr, 0, {"estimate_mode": mode, "conf_target": 0.1, "add_inputs": True}))
+
+        self.log.info("Test walletcreatefundedpsbt with invalid conf_target settings")
+        for mode in ["unset", "economical", "conservative", "btc/kb", "sat/b"]:
+            self.log.debug("{}".format(mode))
+            for k, v in {"string": "", "object": {"foo": "bar"}}.items():
+                assert_raises_rpc_error(-3, "Expected type number for conf_target, got {}".format(k),
+                    lambda: self.nodes[1].walletcreatefundedpsbt(inputs, addr, 0, {"estimate_mode": mode, "conf_target": v, "add_inputs": True}))
+            if mode in ["btc/kb", "sat/b"]:
+                assert_raises_rpc_error(-3, "Amount out of range",
+                    lambda: self.nodes[1].walletcreatefundedpsbt(inputs, addr, 0, {"estimate_mode": mode, "conf_target": -1, "add_inputs": True}))
+                assert_raises_rpc_error(-4, "Fee rate (0.00000000 BTC/kB) is lower than the minimum fee rate setting (0.00001000 BTC/kB)",
+                    lambda: self.nodes[1].walletcreatefundedpsbt(inputs, addr, 0, {"estimate_mode": mode, "conf_target": 0, "add_inputs": True}))
```

Comment:
Zero often being some sort of special case, what happens with values between 0 and 0.00001000? Maybe we should also test `0.00000999`.

Reply:
Agreed--started doing just that today for the new arg [`0f2eb16` (#20231)](https://github.com/bitcoin/bitcoin/pull/20231/commits/0f2eb167ddfa7d2612708d09e05189f969699121#diff-047f185450ff2093df390b522db9b6aa1c87353abf7e7a99c00390db8c707f6eR375), will update here.

Reply:
Added
```python
for unit, feerate in {"SAT/B": 0.99999999, "BTC/KB": 0.00000999}.items():
    self.log.info("- raises RPC error 'fee rate too low' if conf_target {} and estimate_mode {} are passed".format(feerate, unit))
    assert_raises_rpc_error(-4, "Fee rate (0.00000999 BTC/kB) is lower than the minimum fee rate setting (0.00001000 BTC/kB)",
        lambda: self.nodes[1].walletcreatefundedpsbt(inputs, outputs, 0, {"estimate_mode": unit, "conf_target": feerate, "add_inputs": True}))
```